### PR TITLE
Fix charged/rigid simulations

### LINKED ISSUE #138: Variations in density depending on calculation method
I ran into some confusion with the tail correction where I was getting different densities based on how the density is calculated--see snippet for example. Which method is the most correct?

Codes snippet: (uses https://github.com/mosdef-hub/reproducibility_study/pull/137)
```python
import unyt as u
import numpy as np
from pymbar.timeseries import subsampleCorrelatedData as subsample          
                                                                            
import reproducibility_project.src.analysis.equilibration as eq  
from reproducibility_project.src.engines.hoomd.project import clean_data

system_mass = 14436.0 * u.amu  # job.sp.mass * u.amu * job.sp.N_liquid
data = np.genfromtxt(logfile, names=True)
data = clean_data(data)
volume = data["volume"] * u.nm**3
density = np.mean(system_mass/volume).to("g/cm**3")
density2 = (system_mass/np.mean(volume)).to("g/cm**3")
                                           
iseq, _, _, _ = eq.is_equilibrated(volume)                               
if iseq:                                                                    
    uncorr, i, g, N = eq.trim_non_equilibrated(volume)                   
    indices = subsample(uncorr, g=g, conservative=True)                     
    pymbar_volume= volume[indices]
    pymbar_density = (system_mass/np.mean(pymbar_volume)).to("g/cm**3")
    pymbar_density2 = np.mean(system_mass/pymbar_volume).to("g/cm**3")
print(pymbar_density, pymbar_density2, density, density2)
```
```
0.3598699603928832 g/cm**3 0.3748876935629535 g/cm**3 0.37496358408470365 g/cm**3 0.3604007295859862 g/cm**3
```

Attached logfile from job 6bb57f05676b7dc09c020295463f8846 in lrc_shift_subproject
[log-npt.txt](https://github.com/mosdef-hub/reproducibility_study/files/7367694/log-npt.txt)

### PR DESCRIPTION
**Pull request by jennyfothergill**
- Fix averaging by setting t0 (fixes #138) 
- Adjust resolution and order parameters for pppm
- Add rigid constraints for water (Are any other molecules rigid?)
- Add post-processing operation to add `density` and change `kinetic_temperature` to `temperature`.

Big thanks to everyone who helped out with the PPPM debugging, in particular @CalCraven @joaander!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,10 +3,13 @@
 *.png
 *.xyz
 *.pdb
+*.ipynb
 
 # Signac
 .signac_sp_cache.json.gz
 signac_project_document.json
+signac_statepoints.json
```
Comment (justinGilmer) [inline .gitignore:11]: Is it supposed to be `statepoints` with an “s”?

File: reproducibility_project/src/engines/hoomd/project.py [hunk 2]
```diff
@@ -70,10 +72,41 @@ def check_equilibration_npt(job):
...
 
 
+@Project.operation.with_directives({"executable": "$MOSDEF_PYTHON", "ngpu": 1})
+@Project.pre(lambda j: j.sp.engine == "hoomd")
+@Project.pre(lambda j: j.doc.get("nvt_eq"))
+@Project.post(lambda j: j.doc.get("post_processed"))
+def post_process(job):
+    """Run post-processing on the log files."""
+    from shutil import copy
+
```
Comment (justinGilmer) [inline reproducibility_project/src/engines/hoomd/project.py:86]: Just a comment that I did not know about this module. Super cool!!

**Review by justinGilmer — APPROVED** [review]
Small comment but LGTM!

File: .gitignore (hunk 1 above)
Comment (jennyfothergill) [reply .gitignore:11]: It is on my end! When you run init.py, do you get a differently named file?

File: reproducibility_project/src/engines/hoomd/project.py (hunk 2 above)
Comment (jennyfothergill) [reply reproducibility_project/src/engines/hoomd/project.py:86]: I didn't know about it either, but `recfunctions` is AWESOME!

File: .gitignore (hunk 1 above)
Comment (justinGilmer) [reply .gitignore:11]: It did not appear to do that for me. Maybe there is a specific command to export all statepoints to a single json? Either way, that is fine.

**Comment by justinGilmer** [comment]
@tcmoore3 no major rush, whenever you get a chance could you give the hoomd rigid section a quick once over? I am not very familiar with the rigid bodies in hoomd.

**Comment by jennyfothergill** [comment]
No rush to merge this--I may still make some minor adjustments

**Review by tcmoore3 — APPROVED** [review]
Looks good!

**Pull request merged**